cmd/tailscale/cli/cli_test.go

```diff
@@ -1659,3 +1659,51 @@ func TestSingleUseStringFlag(t *testing.T) {
 		})
 	}
 }
+
+func TestSingleUseStringFlagsSetTwice(t *testing.T) {
+	tests := []struct {
+		name     string
+		flagName string
+		vals     []string
+		setup    func(*upArgsT) flag.Value
+	}{
+		{
+			name:     "advertise-tags",
+			flagName: "advertise-tags",
+			vals:     []string{"tag:foo", "tag:bar"},
+			setup: func(u *upArgsT) flag.Value {
+				return &u.advertiseTags
+			},
+		},
+		{
+			name:     "advertise-routes",
+			flagName: "advertise-routes",
+			vals:     []string{"10.0.0.0/8", "192.168.0.0/16"},
+			setup: func(u *upArgsT) flag.Value {
+				return &u.advertiseRoutes
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var upArgs upArgsT
+			fs := flag.NewFlagSet("test", flag.ContinueOnError)
+			fs.SetOutput(io.Discard)
```

Comment:
This exercises your new code, but not necessarily how your new code was wired into the CLI as users will use it.

Reply:
Adding to existing tests that exercise the CLI as wired was complicated by https://github.com/tailscale/tailscale/blob/main/cmd/tailscale/cli/cli.go#L53-L61. I'll take this further.

Reply:
See what I did in my other PR